core/app/models/spree/variant.rb

```diff
@@ -245,7 +245,11 @@ def sku_and_options_text
     end
 
     def in_stock?
-      Rails.cache.fetch(in_stock_cache_key) do
```

Comment:
@Oldharlem I think we can have the same cache key as it is now, but we should also add 

```ruby
after_update :clear_in_stock_cache
```
near https://github.com/spree/spree/blob/master/core/app/models/spree/variant.rb#L57

Reply:
I don't think that would be necessary since the cache_versioning takes care of that. 

In the current situation while cache versioning is disabled the cache entry will be deleted `after_touch` when the stock_item touches the variant after an update through its [`after_save`](https://github.com/spree/spree/blob/382019fe0b6dc3e47fbec98c8b5984b8b88c6e3b/core/app/models/spree/stock_item.rb#L24) callback. 

When cache versioning is enabled the version addition to the [`cache.fetch`](https://github.com/spree/spree/blob/a97fc2476ab2f3d37135bc242364c5e26dc6899b/core/app/models/spree/variant.rb#L252) method call will ensure the `in_stock` state will be reevaluated.